.github/workflows/canary.yml

```diff
@@ -0,0 +1,92 @@
+name: Canary
+on:
+  pull_request:
+    branches-ignore:
+      - "master"
+env:
+  GITHUB_TOKEN: ${{ secrets.ADMIN_TOKEN }}
+  GH_TOKEN: ${{ secrets.ADMIN_TOKEN }}
+  NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+    name: Build
+    if: "!contains(github.event.head_commit.message, 'ci skip') && !contains(github.event.head_commit.message, 'skip ci')"
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          token: ${{ secrets.ADMIN_TOKEN }}
+
+      - name: Prepare repository
+        run: git fetch --unshallow --tags
+
+      - name: Setup node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          cache: 'yarn'
+
+      - name: Install dependencies
+        run: |
+          yarn -v
+          yarn install --immutable
+
+      # - name: Apply semver updates (if any)
+      #   run: |
+      #     if yarn auto version; then
+      #       AUTO_VERSION=$(yarn auto version)
+      #       echo "Auto Version: $AUTO_VERSION"
+      #       if [[ $AUTO_VERSION == pre* ]]; then
+      #         yarn lerna version "$AUTO_VERSION" --no-git-tag-version --no-push --yes --preid "$GITHUB_REF"
+      #       elif [[ $AUTO_VERSION ]]; then
+      #         yarn lerna version "$AUTO_VERSION" --no-git-tag-version --no-push --yes
+      #       fi
+      #     else
+      #       echo "Unable to bump version successfully." && exit 1
+      #     fi
+
+      - name: Build
+        run: yarn build
+
+      - name: Upload Build Results
+        uses: actions/upload-artifact@v3
+        with:
+          name: build-output
+          path: |
+            packages/@magic-ext/*/dist
+            packages/@magic-sdk/*/dist
+            packages/*/dist
```

Comment:
@maksymivash @davidx these artifacts will be uploaded in the current build context, correct? As in, they will not be published to our public npm packages?

Reply:
Unfortunately it looks like that would be uploaded to the pub repo

Reply:
@bcleary06 Canary version should have [these artifacts](https://www.npmjs.com/package/@magic-sdk/provider/v/13.6.2-ef22383-1?activeTab=code) available on NPM no?

Reply:
@bcleary06 @Ariflo Ignore that, action is actually related to sharing the [artifacts in the build context](https://github.com/actions/upload-artifact)